### DISCUSSION TITLE
Error: variant matching problem while running certain PGS IDs

### DISCUSSION THREAD

**Discussion by SoumileeChaudhuri** [PGScatalog/pgsc_calc#117] [Q&A]
Hi, 
I have been using the pgsc_calc workflow to generate PRSs for certain PGS IDs and traits; some of my analyses worked fine but I have been getting mostly errors or confusing results. I have delineated them below but any kind of troubleshooting/technical help would be very appreciated:

1. For PGS calculations that ran successfully, even when my target sample is > 1000, I am having a PGS_AVG column. The guide says I should only have that if my target sample has low size. I still havent been able to figure out whats going on. For example-- PGS IDs PGS002682,PGS001335, and PGS002316 all run but have this problem. 

While PGS002361 which belongs to the same study and population as the previously mentioned PGS-IDs, errors out, which brings me to my second point--> 

2. Some PGS calculations dont run at all, error out and display the same error-- even when they are from different traits. So for example-- PGS000117, PGS00863, (CVD traits) and PGS003355 ( CAD trait)  error out while running and gives the same errors and I am not able to interpret what is wrong. 
I have checked chromosome compatibility, and everything mentioned in the troubleshooting guide like changing overlap parameters but getting the same errors. 
Its something about variants not matching but disappears quickly in the terminal for me to even screenshot it. But its the same error for multiple random PGS IDs, I dont know why some work and some dont. And bing new to this pipeline, i am unable to figure out. 

Would appreciate any help!

**Comment by nebfield** [comment]
Thanks for your question, I'm sorry you're having so much trouble.

We provide `{pgs_id}_AVG` and `{pgs_id}_SUM` columns for each PGS ID in the `aggregated_scores.txt.gz` file. `_AVG` can be better for smaller sample sizes, but the documentation isn't as clear as it should be, so I can see why you're confused. You can choose which score you prefer to use for downstream analysis.

The best way to understand why some scores don't work is to include one score that definitely works with your bad scores, then check the following in the output results directory (normally `results/`):

* the HTML report
* the summary report  
* the full matching log 

We try to explain exactly what's wrong in these files, but they can be tricky to find when no scores work at all (they are made in the`work/` directory until the workflow completes successfully and publishes).

**Reply by SoumileeChaudhuri** [reply]
Thank you! Q1 is very clear to me now. And I will try going through the results by including one good and bad score to see whats going on. It mostly tells me to check nextflow.log for errors and its hard for me to locate any error there. And maybe this time when I can recreate the error, I will try to save it for furthur follow-up. Thanks a lot for this help!
